protocol/src/main/java/org/fido/iot/protocol/To2ServerService.java

```diff
@@ -152,13 +152,13 @@ protected void doProveDevice(Composite request, Composite reply) {
     getStorage().continued(request, reply);
   }
 
-  protected void doAuthDone(Composite request, Composite reply) {
+  protected void doDeviceServiceInfoReady(Composite request, Composite reply) {
     getStorage().continuing(request, reply);
     Composite body = request.getAsComposite(Const.SM_BODY);
     Composite message = Composite.fromObject(getCryptoService().decrypt(body,
         getStorage().getOwnerState()));
 
-    Object hmacObj = message.get(Const.FIRST_KEY);
+    Object hmacObj = message.getAsComposite(Const.FIRST_KEY).get(Const.FIRST_KEY);
```

Comment:
This was correct as is, as per message structure, i.e the 1st key in the message is hmac. Additionally, now the 2nd key directly would be maxOwnerServiceInfoSz. Please update the lines below for the same.

Reply:
Updated.